### PR TITLE
Added a new DAG and tests for MaxDiffusion

### PR DESCRIPTION
**Pull request by parambole**
# Description

Added a new DAG and tests for MaxDiffusion

# Tests

Tested on a dev airflow instance

image

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/imagegen_devx/configs/gke_config.py [hunk 1]
```diff
@@ -25,24 +25,28 @@
     "v4-8": TpuVersion.V4,
     "v4-16": TpuVersion.V4,
     "v5-8": TpuVersion.V5P,
+    "v6e-256": TpuVersion.TRILLIUM,
```
Comment (RissyRan) [inline dags/imagegen_devx/configs/gke_config.py:28]: Sorry I may miss this file a while back, but why not re-use what we have in `vm_resource.py`? We share all GCE and GKE resources in `vm_resource` file.

File: dags/imagegen_devx/maxdiffusion_e2e.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import datetime
+from airflow import models
+from dags import composer_env, test_owner, gcs_bucket
+from dags.vm_resource import Project, TpuVersion, CpuVersion, Zone, DockerImage, GpuVersion, ClusterName
+from dags.imagegen_devx.configs import gke_config as config
+from xlml.utils import name_format
+
+# Run once a day at 4 am UTC (8 pm PST)
+SCHEDULED_TIME = "0 4 * * *" if composer_env.is_prod_env() else None
```
Comment (RissyRan) [inline dags/imagegen_devx/maxdiffusion_e2e.py:26]: Nit: ensure you have manually checked existing runs to avoid traffic for scheduling.

Comment (parambole) [reply dags/imagegen_devx/maxdiffusion_e2e.py:26]: @RissyRan I didn't quite catch that. Can you please elaborate on it ?

File: dags/imagegen_devx/configs/gke_config.py (hunk 1 above)
Comment (parambole) [reply dags/imagegen_devx/configs/gke_config.py:28]: @RissyRan We are using the resources defined in vm_resource. This file simply maintains the mapping, which helps resolve clusters, device types, etc., in test cases.

Comment (RissyRan) [reply dags/imagegen_devx/configs/gke_config.py:28]: Cool, this also makes as it's under imagegen_devx folder, not in general.

File: dags/imagegen_devx/maxdiffusion_e2e.py (hunk 2 above)
Comment (RissyRan) [reply dags/imagegen_devx/maxdiffusion_e2e.py:26]: You could check this at glance on [UI](https://screenshot.googleplex.com/6t658C5npds23w2). So, we won't put all dag runs at 4am UTC ;)

**Review by RissyRan — APPROVED** [review]

File: dags/vm_resource.py [hunk 3]
```diff
@@ -173,7 +171,7 @@ class ClusterName(enum.Enum):
   V5E_16_MULTISLICE_CLUSTER = "v5e-16-bodaborg"
   V5E_256_MULTISLICE_CLUSTER = "v5e-256-bodaborg"
   V5E_256_US_WEST_4_MULTISLICE_CLUSTER = "v5e-256-bodaborg-us-west4"
-  BODABORG_V6E_256_EUROPE_WEST4_A = "bodaborg-v6e-256-europe-west4-a"
+  BODABORG_V6E_256 = "bodaborg-v6e-256"
   A3_CLUSTER = "maxtext-a3-20n"
```
Comment (jonb377) [inline dags/vm_resource.py:175]: Similar change in https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/393, cc @raymondzouu

**Review by jonb377 — APPROVED** [review]
LGTM!

File: dags/vm_resource.py (hunk 3 above)
Comment (parambole) [reply dags/vm_resource.py:175]: @raymondzouu I am making the changes in this PR as my tests weren't working on the previously used cluster.

**Pull request merged**